http/declaration.d.ts

```diff
@@ -0,0 +1,9 @@
+declare module 'superagent-mock' {
```

Comment:
I'm not happy myself about this file, and this was for me the hardest thing to actually make. I had trouble finding any doc on how to do it, went with somehow seeing how `@types/superagent` was made, and imitated some stuff into a `declare module 'superagent-mock' {}` scope ... It worked but no sure on how to clean this up.

Reply:
That's fine, these files don't need to be perfect since they are just placeholders for the lack of proper .d.ts files. Could you @Sinewyk rename it to `superagent-mock.d.ts`?